### PR TITLE
feat(artifacts): adds a visual explanation

### PR DESCRIPTION
**Pull request by lwander**
FYI the svgs are hard to see in the diff view, they look roughly like this: 
uykorqruobb

### CODE REVIEW

File: reference/artifacts/in-pipelines/index.md [hunk 1]
```diff
@@ -135,3 +135,79 @@ whose execution history to search, and an expected artifact to bind.
 
 A common use case would be to "promote" the image deployed to staging to a
 pipeline that's deploying to production.
+
+# A Visual Explanation
+
+To help explain how artifacs & expected artifacts work, let's walk through a
```
Comment (dorbin) [inline reference/artifacts/in-pipelines/index.md]: sp: "artifacts"

File: reference/artifacts/in-pipelines/index.md [hunk 2]
```diff
@@ -135,3 +135,79 @@ whose execution history to search, and an expected artifact to bind.
...
+
+{%
+   include
+   figure
+   image_path="./configuration.svg"
+   caption="The pipeline declares that it expectes an artifact matching _1_
+   (perhaps a docker image) _when the pipeline starts_ in the pipeline
+   configuration, and an artifact matching _2_ in pipeline stage _B_ (perhaps a
+   \"Find Artifact from Execution\" stage)."
+%}
```
Comment (dorbin) [inline reference/artifacts/in-pipelines/index.md]: Sentence is a little awkward. Is it enough to just say "...when the pipeline starts"?

File: reference/artifacts/in-pipelines/index.md [hunk 3]
```diff
@@ -135,3 +135,79 @@ whose execution history to search, and an expected artifact to bind.
...
+
+When the pipeline starts, it is triggered by some source (maybe a Webhook)
```
Comment (dorbin) [inline reference/artifacts/in-pipelines/index.md]: Cause/effect? 
Maybe "The pipeline starts when it's triggered by..."

File: reference/artifacts/in-pipelines/index.md [hunk 4]
```diff
@@ -135,3 +135,79 @@ whose execution history to search, and an expected artifact to bind.
...
+
+{%
+   include
+   figure
+   image_path="./running-b-d.svg"
+%}
+
+If stages _C_ or _D_ needed to reference an upstream artifact, they would have
+different artifacts accessible to them, since they have different upstream
+stages.
```
Comment (dorbin) [inline reference/artifacts/in-pipelines/index.md]: "For example, stage D does not have access to artifact 2." 
right?

**Review by dorbin — APPROVED** [review]

File: reference/artifacts/in-pipelines/index.md (hunk 4 above)
Comment (lwander) [reply reference/artifacts/in-pipelines/index.md]: yup!

**Pull request merged**